File: CPP/Code/Spesifikasi.cpp
#include <iostream>
#include "Detail.cpp"
using namespace std;

class Spesifikasi : public Detail{

	//atribut pakenya private
	private:
		int harga;
		string garansi;
		string tahunproduksi;

	//method yang bersifat public
	public:
    
		//constructor untuk inisiasi atribut
        Spesifikasi(string nama, string kode, string merk, int stok, int daya, double berat, int harga, string garansi, string tahunproduksi) : Detail(nama, kode, merk, stok, daya, berat) {
            this->harga = harga;
            this->garansi = garansi;
            this->tahunproduksi = tahunproduksi;
	    }

		//getter harga
		void setHarga(int harga) {
			this->harga = harga;
		}
		
		//setter harga
		int getHarga() {
			return this->harga;
		}
		
		//getter garansi
		void setGaransi(string garansi) {
			this->garansi = garansi;
		}
		
		//setter garansi
		string getGaransi() {
			return this->garansi;
		}
		
		//getter tahunproduksi
		void setTahunproduksi(string tahunproduksi) {
			this->tahunproduksi = tahunproduksi;
		}
		
		//setter tahunproduksi
		string getTahunproduksi() {
			return this->tahunproduksi;
		}
		
		~Spesifikasi(){ }

};

File: PHP/Code/Spesifikasi.php
<?php

//manggil detail.php
require_once "Detail.php";

class Spesifikasi extends Detail {

    //atributs
    private int $harga;
    private string $garansi;
    private string $tahunproduksi;
    private string $gambar;

    public function __construct(string $nama, string $kode, string $merk, int $stok, int $daya, float $berat, int $harga, string $garansi, string $tahunproduksi, string $gambar) {
        parent::__construct($nama, $kode, $merk, $stok, $daya, $berat);
        $this->setHarga($harga);
        $this->garansi = $garansi;
        $this->tahunproduksi = $tahunproduksi;
        $this->gambar = $gambar;
    }

    //getter setter atributs
    public function getHarga(): int { 
        return $this->harga;
    }
    public function setHarga(int $harga): void { 
        $this->harga = ($harga < 0) ? 0 : $harga; 
    }

    public function getGaransi(): string { 
        return $this->garansi; 
    }
    public function setGaransi(string $garansi): void { 
        $this->garansi = $garansi; 
    }

    public function getTahunproduksi(): string { 
        return $this->tahunproduksi; 
    }
    public function setTahunproduksi(string $tahunproduksi): void { 
        $this->tahunproduksi = $tahunproduksi; 
    }

    public function getGambar(): string { 
        return $this->gambar;
     }
    public function setGambar(string $gambar): void { 
        $this->gambar = $gambar; 
    }
}
